parastell/invessel_build.py

```diff
@@ -624,8 +642,8 @@ def wall_s(self, s):
             self._logger.error(e.args[0])
             raise e
         
-        if hasattr(self, 'radial_build'):
-            self.radial_build = self.radial_build
+        if hasattr(self, 'separate_chamber'):
+            self.separate_chamber = self.separate_chamber
```

Comment:
Don't we always have this attribute now?  

Also, is this construct typical?  It seems confusing to me since it looks like it just sets the variable equal to itself.  I know it is calling the setter, and therefore doing other things.  I wonder if we move the actual code of the setter into some other function (`update_separate_chamber` or something) and call it form both the setter and directly here?

Reply:
I'm not sure how typical it is. However, I agree that having a function for this would clarify what's going on, rather than just calling the setter.

Reply:
While true that the class will always have the `separate_chamber` attribute, the `hasattr` condition there for the purpose of checking whether or not the attribute has been set yet. If the class is being initialized, `separate_chamber` will not yet have been set, and so I don't want the code to call the setter.

However, if `wall_s` is switched after initialization, I want the code to update `sol` in `radial_build` accordingly, which is handled by the `separate_chamber` setter.